content/posts/2021/10/libplanet-0.11-0.19/index.eng.md

```diff
@@ -0,0 +1,251 @@
+[`NonblockRenderer<T>`][NonblockRenderer<T>] and
+[`NonblockActionRenderer<T>`][NonblockActionRenderer<T>] classes,
+are introduced.  Both classes take another render through their constructors,
+and call its event handlers in a separated thread.  As they are decorators,
+the existing renderers can be transformed into a non-blocking style,
+by wrapping existing ones with the decorators.
+
+Despite they are non-blocking, they have their own internal queue to guarantee
+the order of events.
+
+[IRenderer<T>]: https://docs.libplanet.io/0.19.0/api/Libplanet.Blockchain.Renderers.IRenderer-1.html
+[IActionRenderer<T>]: https://docs.libplanet.io/0.19.0/api/Libplanet.Blockchain.Renderers.IActionRenderer-1.html
+[NonblockRenderer<T>]: https://docs.libplanet.io/0.19.0/api/Libplanet.Blockchain.Renderers.NonblockRenderer-1.html
+[NonblockActionRenderer<T>]: https://docs.libplanet.io/0.19.0/api/Libplanet.Blockchain.Renderers.NonblockActionRenderer-1.html
+
+
+Configurable Transaction Preference By Miners
+---------------------------------------------
+
+*Added in version [0.17.0].*
+
+The protocol does not require miners to have any priority for transactions
+to include in a block, if these staisfy
+[`IBlockPolicy<T>.ValidateNextBlockTx()` predicate
+method][IBlockPolicy<T>.ValidateNextBlockTx] at least.[^3]  However, if there
+are too many transactions to be included in a block, a miner may need to
+prioritize them so that they are chunked into multiple blocks.
+
+Since version 0.17.0, instead of Libplanet arbitrarily prioritizing them,
+miners can configure the priority for transactions to include in a block.
+API-wise, [`BlockChain<T>.MineBlock()` method][BlockChain<T>.MineBlock] has the
+`txPriority` option which takes an [`IComparer<Transaction<T>>`][IComparer<T>].
+
+[^3]: Of course, there is an order between transactions signed by a signer.
+      These are ordered by [`Transaction<T>.Nonce`][Transaction<T>.Nonce].
+
+[IBlockPolicy<T>.ValidateNextBlockTx]: https://docs.libplanet.io/0.19.0/api/Libplanet.Blockchain.Policies.IBlockPolicy-1.html#Libplanet_Blockchain_Policies_IBlockPolicy_1_ValidateNextBlockTx_Libplanet_Blockchain_BlockChain__0__Libplanet_Tx_Transaction__0__
+[Transaction<T>.Nonce]: https://docs.libplanet.io/0.19.0/api/Libplanet.Tx.Transaction-1.html#Libplanet_Tx_Transaction_1_Nonce
+[BlockChain<T>.MineBlock]: https://docs.libplanet.io/0.19.0/api/Libplanet.Blockchain.BlockChain-1.html#Libplanet_Blockchain_BlockChain_1_MineBlock_Libplanet_Crypto_PrivateKey_DateTimeOffset_System_Boolean_System_Int32_System_Int32_IComparer_Libplanet_Tx_Transaction__0___CancellationToken_
+[IComparer<T>]: https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.icomparer-1
+
+
+Blocks Signed By Miners
+-----------------------
+
+*Added in version [0.18.0].*
+
+Since the protocol version 2, i.e., Libplanet 0.18.0, every block is signed with
+the private key of its miner, and it includes the public key besides the addres
+of its miner.
```

Comment:
```suggestion
Since protocol version 2, i.e., Libplanet 0.18.0, every block is signed with
the private key of its miner, and it includes the public key as well as the address
of its miner.
```